src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/structure/OverflowDbNode.java

```diff
@@ -450,11 +461,17 @@ public int occurrenceToBlockOffset(Direction direction,
 
     @Override
     public boolean hasNext() {
+      // there may be holes, e.g. if an edge was removed
+      while (current < exclusiveEnd) {
```

Comment:
Not required. We should deem all iterators invalid as soon as edges get added/removed.

Reply:
you can see that it's required by deleting it and running e.g. `removeEdgeSimple` test. 
scenario: adding one edge, removing it again. `hasNext` will return `true`, even though there is no edge. that's because we leave 'holes' in the array when removing edges. 
in a later step we could defragment the array, but we probably don't want to do that for every single remove operation, so we'd need this check anyway

Reply:
You are right. And defragmentation we will need when we do lots of remove and add operations.

Reply:
yup. first step is simple though: re-fill the gaps when adding a new edge. will do as a follow up